pkg/middleware/short_url_redirect.go

```diff
@@ -0,0 +1,29 @@
+package middleware
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+	"gopkg.in/macaron.v1"
+)
+
+func ShortUrlRedirect() macaron.Handler {
+	return func(res http.ResponseWriter, req *http.Request, c *macaron.Context) {
+		shortUrlUid := c.Params(":uid")
+		ctx := c.Data["ctx"].(*models.ReqContext)
+
+		if !util.IsValidShortUID(c.Params(":uid")) {
+			return
+		}
+
+		cmd := models.GetFullUrlQuery{OrgId: ctx.OrgId, Uid: shortUrlUid}
+		if err := bus.Dispatch(&cmd); err != nil {
+			return
+		}
+		c.Redirect(setting.ToAbsUrl(strings.TrimPrefix(cmd.Result.Path, "/")), 302)
```

Comment:
Trimming of prefix seems like it rather should be done in the write path, when creating a short url.